### PR TITLE
Fixes #34071 - Stop using deprecated redux toast actions

### PR DESCRIPTION
**Pull request by ofedoren**
#### What are the changes introduced in this pull request?
Removing of deprecated toast actions in favor of a new way of using toats.
#### What are the testing steps for this pull request?
Updated tests/snapshots, but please, make sure I didn't break anything.

### CODE REVIEW

File: webpack/__mocks__/foremanReact/redux/actions/toasts.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export const addToast = toast => ({
```
Comment (tbrisker) [inline webpack/__mocks__/foremanReact/redux/actions/toasts.js:1]: This file should be renamed to `webpack/__mocks__/foremanReact/components/ToastsList/index`

Comment (ofedoren) [reply webpack/__mocks__/foremanReact/redux/actions/toasts.js:1]: :facepalm: sure, done.

**Comment by tbrisker** [comment]
@ofedoren looks like the snapshots were generated with the real action (that adds a key and has a bit different structure to the mock), but in CI the mock is being used. Yet another example showing why snapshot tests are mostly useless...

File: webpack/utils/helpers.js [hunk 2]
```diff
@@ -81,6 +81,7 @@ export const sendErrorNotifications = messages => (dispatch) => {
     const message = typeof msg === 'string' ? msg : `${msg.message}: ${msg.details}`;
     dispatch(addToast({
       type: 'error',
+      key: 'toastError',
```
Comment (tbrisker) [inline webpack/utils/helpers.js:84]: Could there be more than on error thrown at the same time? not sure we should use a fixed key here.

Comment (ofedoren) [reply webpack/utils/helpers.js:84]: Well, I'm not sure either. But I'd find a bad UX in multiple toasts thrown at me with error. On the other hand, it's possible that different components would call different APIs and each of them fails with a toast... :/

Comment (tbrisker) [reply webpack/utils/helpers.js:84]: Indeed, they might even have different error messages

Comment (Andrewgdewar) [reply webpack/utils/helpers.js:84]: this seems like a good spot to use the forEach((msg, index) => 
... 
key: `toastError${index}`,
... 
and disable the eslint rule around such things for this line. no?

File: webpack/__mocks__/foremanReact/components/ToastsList/index.js [hunk 3]
```diff
@@ -1,7 +1,8 @@
 export const addToast = toast => ({
```
Comment (Andrewgdewar) [inline webpack/__mocks__/foremanReact/components/ToastsList/index.js:1]: Why are you moving this file into mocks? is this intentional

**Comment by Andrewgdewar** [comment]
So after testing this PR against the latest foreman develop, I am unable to start katello, I am getting the following errors. 
Screen Shot 2021-12-02 at 12 09 00 PM

**Comment by Andrewgdewar** [comment]
Upon removing both references to the issues seen above, I can now load into katello, but I am seeing the following foreman-task gem failure
Screen Shot 2021-12-02 at 12 37 21 PM
es

adding the following line to update to the latest foreman-tasks in foreman was necessary to get the application to work.

**Comment by Andrewgdewar** [comment]
Closing this, have opened another PR [here](https://github.com/Katello/katello/pull/9829) to address these changes.
Note: A PR/release updating foreman-tasks is needed to prevent katello failures.

**Pull request closed without merging**